packages/camera/camera_avfoundation/ios/camera_avfoundation/Sources/camera_avfoundation/DefaultCamera.swift

```diff
@@ -23,10 +23,182 @@ final class DefaultCamera: FLTCam, Camera {
 
   /// Maximum number of frames pending processing.
   /// To limit memory consumption, limit the number of frames pending processing.
-  /// After some testing, 4 was determined to be the best maximuńm value.
+  /// After some testing, 4 was determined to be the best maximum value.
   /// https://github.com/flutter/plugins/pull/4520#discussion_r766335637
   private var maxStreamingPendingFramesCount = 4
 
+  private var exposureMode = FCPPlatformExposureMode.auto
+  private var focusMode = FCPPlatformFocusMode.auto
+
+  func reportInitializationState() {
+    // Get all the state on the current thread, not the main thread.
+    let state = FCPPlatformCameraState.make(
+      withPreviewSize: FCPPlatformSize.make(
+        withWidth: Double(previewSize.width),
+        height: Double(previewSize.height)
+      ),
+      exposureMode: exposureMode,
+      focusMode: focusMode,
+      exposurePointSupported: captureDevice.isExposurePointOfInterestSupported,
+      focusPointSupported: captureDevice.isFocusPointOfInterestSupported
+    )
+
+    FLTEnsureToRunOnMainQueue { [weak self] in
+      self?.dartAPI?.initialized(with: state) { _ in
+        // Ignore any errors, as this is just an event broadcast.
+      }
+    }
+  }
+
+  func receivedImageStreamData() {
+    streamingPendingFramesCount -= 1
+  }
+
+  func start() {
+    videoCaptureSession.startRunning()
+    audioCaptureSession.startRunning()
+  }
+
+  func stop() {
+    videoCaptureSession.stopRunning()
+    audioCaptureSession.stopRunning()
+  }
+
+  func setExposureMode(_ mode: FCPPlatformExposureMode) {
+    exposureMode = mode
+    applyExposureMode()
+  }
+
+  private func applyExposureMode() {
+    try? captureDevice.lockForConfiguration()
```

Comment:
can we do `defer { captureDevice.unlockForConfiguration() }` in case we update the code and introduce an early exit.